workitem_blackbox_test.go

```diff
@@ -900,13 +913,47 @@ func (s *WorkItem2Suite) TestWI2ListByAssigneeFilter() {
 	assert.Equal(s.T(), newUser.ID.String(), *wi.Data.Relationships.Assignees.Data[0].ID)
 
 	newUserID := newUser.ID.String()
-	_, list := test.ListWorkitemOK(s.T(), s.svc.Context, s.svc, s.wi2Ctrl, nil, &newUserID, nil, nil)
+	_, list := test.ListWorkitemOK(s.T(), s.svc.Context, s.svc, s.wi2Ctrl, nil, &newUserID, nil, nil, nil)
 
 	assert.Len(s.T(), list.Data, 1)
 	assert.Equal(s.T(), newUser.ID.String(), *list.Data[0].Relationships.Assignees.Data[0].ID)
 	assert.True(s.T(), strings.Contains(*list.Links.First, "filter[assignee]"))
 }
 
+func (s *WorkItem2Suite) TestWI2ListByIterationFilter() {
+	tempIteration := createOneRandomIteration(s.svc.Context, s.db)
+	require.NotNil(s.T(), tempIteration)
+	iterationID := tempIteration.ID.String()
+	c := minimumRequiredCreatePayload()
+	c.Data.Attributes[workitem.SystemTitle] = "Title"
+	c.Data.Attributes[workitem.SystemState] = workitem.SystemStateNew
+	c.Data.Relationships = &app.WorkItemRelationships{
+		BaseType: &app.RelationBaseType{
+			Data: &app.BaseTypeData{
+				Type: "workitemtypes",
+				ID:   "bug",
+			},
+		},
+		Iteration: &app.RelationGeneric{
+			Data: &app.GenericData{
+				ID: &iterationID,
+			},
+		},
+	}
+	_, wi := test.CreateWorkitemCreated(s.T(), s.svc.Context, s.svc, s.wi2Ctrl, &c)
+	require.NotNil(s.T(), wi.Data)
+	require.NotNil(s.T(), wi.Data.ID)
+	require.NotNil(s.T(), wi.Data.Type)
+	require.NotNil(s.T(), wi.Data.Attributes)
+	require.NotNil(s.T(), wi.Data.Relationships.Iteration)
+	assert.Equal(s.T(), iterationID, *wi.Data.Relationships.Iteration.Data.ID)
+
+	_, list := test.ListWorkitemOK(s.T(), s.svc.Context, s.svc, s.wi2Ctrl, nil, nil, &iterationID, nil, nil)
+	assert.Len(s.T(), list.Data, 1)
+	assert.Equal(s.T(), iterationID, *list.Data[0].Relationships.Iteration.Data.ID)
```

Comment:
You should make the line before require.Len because your program shall stop before it crashes due to accessing the first element of an empty array.